Feature/your results page

ItzDusty opened the pull request:
## Description

This pull request delivers the final “Your Results” page:  
- Full HTML structure for header, breadcrumb nav, search/filter controls, results table, pagination and footer.  
- Complete CSS for responsive header (logo, nav-links, avatar), table styling, footer positioning, and mobile menu.  
- JavaScript to load results (with fallback data), render table pages, wire up search & filters, and toggle the hamburger menu.  
- Ensures the footer circles stay at the bottom without overlapping content.

---

## Related Issue(s)

Closes #21 

---

## Changes Made

- [x] Added HTML markup for header, main content, and footer  
- [x] Wrote CSS (`styles.css`) for flex-based layout, fluid logo/avatar sizing, responsive breakpoints, and sticky footer behavior  
- [x] Implemented JavaScript (`<script>` block) for data fetching, pagination, live search, filters, and mobile nav toggle  
- [-] Accessibility tweaks (aria-labels, focus states)  
- [X] Refactoring / cleanup

---

## Screenshots
![Desktop view](https://github.com/user-attachments/assets/f3b16911-dc83-49dc-8d51-2a7e3db2fd14)
![Mobile view](https://github.com/user-attachments/assets/536c6621-003e-4763-9db9-1339f6703cdc)
![Hamburger menu](https://github.com/user-attachments/assets/b4bfe3fd-0a0f-4b31-9af0-f6f9fa2630a4)
![image](https://github.com/user-attachments/assets/6703e3dc-47a5-41a3-b299-638054690485)
![image](https://github.com/user-attachments/assets/3734e2cf-8d29-439f-8c93-804ac46c6cef)
![image](https://github.com/user-attachments/assets/b1c013a4-75e5-4a8c-ada2-01fcfeebf4b5)






---

## Acceptance Criteria

- [X] Changes have been tested locally  
- [X] Code follows the project’s style guide  
- [X] Page matches the provided design specification  
- [X] No console errors or warnings  
- [X] Fully responsive on desktop and mobile screens

ItzDusty committed:
Initial commit, baisc HTML, CSS and JS set-up for results page

The CSS and HTML set-up for header and footer could be re-used by the other developers on the team, could potentially save someone some headache?

ItzDusty committed:
Merge branch 'develop' into feature/your-results-page

ItzDusty committed:
Final HTML, CSS and JS - Ready for QA

I’ve completed as much of the HTML, CSS, and JavaScript as I could this week. There may still be aspects that need refinement to meet our QA standards.

Haimrich9 reviewed (changes requested):
refer to figma file for mobile design
- Profile icon should be inside the hamburger menu
- Inside the hamburger menu there is missing icons and design shown in figma file
- There is svg files for the arrows  >>  <<  <  > called "last" "first" "left" "right"
- missing icon view_results.png for mobile 
<img width="253" alt="Skjermbilde 2025-05-08 kl  02 33 13" src="https://github.com/user-attachments/assets/e1f08b5d-761e-4cb7-880a-62169bde3aa7" />
<img width="133" alt="Skjermbilde 2025-05-08 kl  02 33 39" src="https://github.com/user-attachments/assets/54ba6f9a-e5ba-472d-b292-c72b11383257" />

ItzDusty committed:
Started refactoring

Refactoring code and fixing last changes before pull request.
Some coloring issues and hamburger menu issues remain.
Will probably finish tomorrow.

ItzDusty committed:
Further fixed CSS and HTML issues

All that remains before completing pull-request;
Fix hamburger menu design, elements must right align, and header must have color change - along with some logo issues.

ItzDusty committed:
Fixed feedback from last pullrequest

I have finalized hamburger menu.
Adjusted design to Figma.
Fixed Hamburgermenu.
Optimized JS.
Optimized CSS.

ItzDusty commented:
@Haimrich9 I have completed your comments. Please let me know if further adjustments are required.

Haimrich9 reviewed (changes requested):
Correct font families according to this: 
<img width="668" alt="Skjermbilde 2025-05-29 kl  15 10 07" src="https://github.com/user-attachments/assets/f3582c59-fcfb-4cf1-b612-4a575b33ac2d" /> The fonts are from google fonts

the gray border here should be #cccccc
<img width="681" alt="Skjermbilde 2025-05-29 kl  15 20 43" src="https://github.com/user-attachments/assets/cc47f8e9-722b-46d8-b777-13b4ea53da2e" />

For this project, please use px for all measurements. Since we as a class haven’t yet covered rem/em, sticking to px will keep things simpler and more consistent for everyone.

make the "back to dashboard" text blue like this:
<img width="585" alt="Skjermbilde 2025-05-29 kl  15 16 39" src="https://github.com/user-attachments/assets/09b2c1b7-fb5f-45be-93f2-18257da7d832" />

Instead of using span to make the hamburger icon, use the already made one in /assets/icons/nav/menu.png

Logo height on desktop should be: 176px. and mobile should be: 64px

Haimrich9 reviewed (changes requested):
add more space between navigation links to match figma design:
<img width="462" alt="Skjermbilde 2025-05-29 kl  15 36 35" src="https://github.com/user-attachments/assets/e091cf92-d4e5-499f-b2ea-1492a81737d4" />

ItzDusty committed:
Fixed latest comments from pull request

ItzDusty commented:
[x] Hamburger menu icon added.
[x] REM and EM replaced with PX according to QA's demand.
[x] Correct fonts imported.
[x] Spacing between links corrected according to figma design.
[x] Logo sizes hardcoded to 176px and 64px rather than responsive sizes, according to QA's demand.

Hopefully we can close this ticket now.

![Desktop view](https://github.com/user-attachments/assets/f3b16911-dc83-49dc-8d51-2a7e3db2fd14)
![Mobile view](https://github.com/user-attachments/assets/536c6621-003e-4763-9db9-1339f6703cdc)
![Hamburger menu](https://github.com/user-attachments/assets/b4bfe3fd-0a0f-4b31-9af0-f6f9fa2630a4)
![image](https://github.com/user-attachments/assets/6703e3dc-47a5-41a3-b299-638054690485)
![image](https://github.com/user-attachments/assets/3734e2cf-8d29-439f-8c93-804ac46c6cef)
![image](https://github.com/user-attachments/assets/b1c013a4-75e5-4a8c-ada2-01fcfeebf4b5)

Haimrich9 reviewed (changes requested):
**Design Fixes:**
   - Table Alignment: 
        - The table should be horizontally centered on the page, matching the Figma design.

   - Navigation Links:
        - Navigation links still do not align with the Figma layout. The spacing between the links needs to be fixed
        
**Responsive Issues:**
   - Hamburger Menu Trigger:
        - The hamburger menu appears too early — it becomes visible before the layout fully switches to the mobile header. when it switches to hamburger menu, the hight of the header should also be switched to mobile size
→ Adjust the media query so the menu only appears after the logo and header have shifted to the mobile layout.
       
<img width="855" alt="Skjermbilde 2025-05-31 kl  01 02 35" src="https://github.com/user-attachments/assets/a24e01b9-6231-4a45-8bfc-f3119fe7f3a7" />
<img width="805" alt="Skjermbilde 2025-05-31 kl  01 04 13" src="https://github.com/user-attachments/assets/212a35ac-4999-4614-b298-b6ed0911e4a4" />

ItzDusty commented:
I'll design according to your requirements - FYI the logo size for desktop is wrong on the student dashboard design (150px and not 176px like you told me) - and from what I can see there's less gap/ space between navigational links than, I'll once again match the navigational links gap according to the figma design.

ItzDusty committed:
Adjusted design according to QA demands.

ItzDusty committed:
Modified last css parts.

ItzDusty commented:
PR Update: Navigation & Table Responsiveness (QA Feedback)
Summary of changes based on QA feedback and Figma alignment:

**Navigation Bar**
- Adjusted spacing between navigation links to match student-dashboard: first link now uses margin-left: auto so links are grouped and spaced correctly.
- Ensured the logo and navigation links are aligned as per design.
- Hamburger menu and mobile logo now display at ≤768px, matching the design spec.
- Navigation/header height is consistent (64px) across desktop and mobile, including the mobile nav panel header.
`header nav a {
  text-decoration: none;
  color: #333333;
  font-weight: bold;
  position: relative;
  padding: 0 30px;
  display: flex;
  align-items: center;
  margin-left: auto;
}`

**Table Responsiveness**
- The table now reduces columns (showing only Subject, Exam, Grade, and info icon) at ≤768px instead of 480px, as required.
- Info icon for row details is now visible at 768px and below.
- Removed any conflicting or redundant 480px table reduction rules.

**General Mobile Improvements**
- Added/adjusted media queries for better appearance and usability on very small screens (below 320px).
- Cleaned up and unified mobile-specific CSS for consistency.

**Result:**
The navigation and results table now match the QA/Figma layout for both desktop and mobile breakpoints, with correct spacing, alignment, and responsive behavior.

<768px: 
![image](https://github.com/user-attachments/assets/1532c709-3ad2-48b0-8676-1120424c9b84)
![image](https://github.com/user-attachments/assets/66379de5-cfe2-4153-b0ce-5e6391e5987a)


>768px:
![image](https://github.com/user-attachments/assets/e693256c-7a72-4fd5-8b23-0512dbc4591d)
![image](https://github.com/user-attachments/assets/5e234120-da7e-4873-a4e8-7e55f0231e19)

ItzDusty committed:
Updated styling.css

ItzDusty commented:
PR Update: Navigation & Table Responsiveness (QA Feedback)
Summary of changes based on QA feedback and Figma alignment:

**Navigation Bar**
- Adjusted spacing between navigation links to match student-dashboard: first link now uses margin-left: auto so links are grouped and spaced correctly.
- Ensured the logo and navigation links are aligned as per design.
- Hamburger menu and mobile logo now display at ≤768px, matching the design spec.
- Navigation/header height is consistent (64px) across desktop and mobile, including the mobile nav panel header.
`header nav a {
  text-decoration: none;
  color: #333333;
  font-weight: bold;
  position: relative;
  padding: 0 30px;
  display: flex;
  align-items: center;
  margin-left: auto;
}`

**Table Responsiveness**
- The table now reduces columns (showing only Subject, Exam, Grade, and info icon) at ≤768px instead of 480px, as required.
- Info icon for row details is now visible at 768px and below.
- Removed any conflicting or redundant 480px table reduction rules.

**General Mobile Improvements**
- Added/adjusted media queries for better appearance and usability on very small screens (below 320px).
- Cleaned up and unified mobile-specific CSS for consistency.

**Result:**
The navigation and results table now match the QA/Figma layout for both desktop and mobile breakpoints, with correct spacing, alignment, and responsive behavior.

<768px: 
![image](https://github.com/user-attachments/assets/1532c709-3ad2-48b0-8676-1120424c9b84)
![image](https://github.com/user-attachments/assets/66379de5-cfe2-4153-b0ce-5e6391e5987a)


>768px:
![image](https://github.com/user-attachments/assets/e693256c-7a72-4fd5-8b23-0512dbc4591d)
![image](https://github.com/user-attachments/assets/5e234120-da7e-4873-a4e8-7e55f0231e19)

Haimrich9 committed:
Merge branch 'develop' into feature/your-results-page

Haimrich9 committed:
Merge branch 'develop' into feature/your-results-page
